### PR TITLE
Fix local frame for surface edges and add terrarium project

### PR DESCRIPTION
**Pull request by pentacular**

### CODE REVIEW

**Comment by pentacular** [comment]
I'm not entirely sure what the local frame of a section ought to be.

I suspect it should retain that of the object being sectioned, in which case flat and friends need to change.

**Comment by BarbourSmith** [comment]
I like upright()

File: nb/api/eachEdge.md [hunk 1]
```diff
@@ -46,3 +46,18 @@ Box(5, 5, 5)
 ![Image](eachEdge.md.1.png)
 
 Box(5, 5, 5).eachEdge((edge, length) => Arc(0.5, [0.2, 0.7], [length]).hasAngle(0 / 4, 2 / 4).to(edge), (edges, face) => edges.and(face.cut(inset(1)).e(0.2)))
+
+```JavaScript
+Box(50, 50, 50)
```
Comment (BarbourSmith) [inline nb/api/eachEdge.md:51]: That's a simple and neat example

**Review by BarbourSmith — APPROVED** [review]

**Comment by hrgdavor** [comment]
this sound cool but I get a blank screen
https://jsxcad.js.org/alpha/#https://raw.githubusercontent.com/jsxcad/JSxCAD/master/nb/projects/pentacular/terrarium/terrarium.nb

there were no errros in console, and only spinner.css failed in network tab

**Pull request merged**